### PR TITLE
feat: allow to configure client id claims

### PR DESCRIPTION
**Pull request by ytvnr**
Fixes https://github.com/gravitee-io/issues/issues/4900

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -41,6 +41,7 @@ Both the header and payload are encoded with Base64, so anyone can read the cont
 |resolverParameter||Needed if you use the `GATEWAY_KEYS` or `GIVEN_ISSUER` resolver (EL support)|string|
 |useSystemProxy||Select this option if you want use system proxy (only useful when resolver is `JWKS_URL`)|boolean|false
 |extractClaims||Select this option if you want to extract claims into the request context|boolean|false
+|clientIdClaim||Needed if the client_id can be find in another claim|string|client_id
```
Comment (brasseld) [inline README.adoc:44]:
```suggestion
|clientIdClaim||Required if client_id should be read from non-standard claims (aud, azp, client_id)|string|client_id
```

File: src/main/java/io/gravitee/policy/jwt/JWTPolicy.java [hunk 2]
```diff
@@ -166,12 +166,14 @@ public void onRequest(Request request, Response response, ExecutionContext execu
     }
 
     private String getClientId(JWTClaimsSet claims) {
-        String clientId = null;
+        String clientId = (String) claims.getClaim(configuration.getClientIdClaim());
```
Comment (brasseld) [inline src/main/java/io/gravitee/policy/jwt/JWTPolicy.java:169]: What if the configured claim is an array?

**Review by brasseld — CHANGES_REQUESTED** [review]

File: src/main/java/io/gravitee/policy/jwt/JWTPolicy.java (hunk 2 above)
Comment (ytvnr) [reply src/main/java/io/gravitee/policy/jwt/JWTPolicy.java:169]: didn't know it was possible.
do you want a mechanism asit's done for audClaim ? 
https://github.com/gravitee-io/gravitee-gateway/pull/644/files#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8R279-R292

File: src/main/java/io/gravitee/policy/jwt/JWTPolicy.java [hunk 3]
```diff
@@ -166,26 +166,21 @@ public void onRequest(Request request, Response response, ExecutionContext execu
     }
 
     private String getClientId(JWTClaimsSet claims) {
-        String clientId = null;
+        Object clientIdClaim = claims.getClaim(configuration.getClientIdClaim());
```
Comment (aelamrani) [inline src/main/java/io/gravitee/policy/jwt/JWTPolicy.java:169]: Do you think that if the client id claim is configured, we should also check the other (default) claims?

I suppose that if you specify a client id claim, you only want it to be tested and not the default claims.

Wdyt?

**Review by aelamrani — CHANGES_REQUESTED** [review]

**Pull request merged**